### ISSUE TITLE
Push to Github and Pypi is not working after migration to GHA

### ISSUE THREAD

**Issue by dgarros** [networktocode/diffsync#93]
### Environment
* DiffSync version:  1.4.0

### Observed Behavior
When we publish a new release the pipeline is failing to publish the new version to Github and Pypi

https://github.com/networktocode/diffsync/actions/runs/1746029264

**Comment by maximumG** [comment]
I don't know if the issue is related but after installing diffsync v1.4.0 from Pypi in a new virtual environment, we are not able to use diffsync anymore.

After some investigation, it seems that _pydantic_ is missing as a requirement in the wheel package published on Pypi. Pydantic will be only install if we install the extras docs packages (`pip install diffsync==1.4.0[docs]`)

```
python3 -m venv 1.4.0
cd 1.4.0
source bin/activate
pip3 install diffsync==1.4.0

Installing collected packages: diffsync
Successfully installed diffsync-1.4.0
# /!\ missing pydantic dependency installation
```

**Comment by dgarros** [comment]
Thank you @maximumG, indeed looks like pydantic has been added to the `docs` extra by mistake
we'll fix that today and publish a 1.4.1
https://github.com/networktocode/diffsync/blob/main/pyproject.toml#L57

I'll open a new issue as well because the original issue (Github/ Pypi) is different

**Comment by dgarros** [comment]
Both issues have been fixed, 
Thank you @ubajze for fixing the CI Pipeline
